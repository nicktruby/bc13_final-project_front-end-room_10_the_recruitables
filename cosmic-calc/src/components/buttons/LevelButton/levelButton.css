.levelButton{
    padding-top: 10%;
}