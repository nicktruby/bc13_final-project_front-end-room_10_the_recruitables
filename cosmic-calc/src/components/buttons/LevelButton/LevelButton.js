import React from "react";
import "./levelButton.css";

function LevelButton({ ButtonNumber, clickToGame }) {
  return (
    <div className="levelButton">
      <button
        className={"levelButtons"}
        onClick={clickToGame}
        id={`levelButton-${ButtonNumber}`}
      >
        {/* BUTTON {ButtonNumber} */}
      </button>
    </div>
  );
}

export default LevelButton;
