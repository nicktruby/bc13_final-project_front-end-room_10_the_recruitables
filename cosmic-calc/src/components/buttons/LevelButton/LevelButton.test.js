import React from "react";
import "@testing-library/jest-dom";
import { screen, render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import LevelButton from "./LevelButton";

describe("<LevelButton>", () => {
  const mockedProps = {
    buttonNumber: "1234",
    clickToGame: jest.fn(),
  };

  it("successfully renders the component, matches snapshot", () => {
    const { asFragment } = render(<LevelButton {...mockedProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("click of button fires the provided 'clickToGame' callback prop", () => {
    render(<LevelButton {...mockedProps} />);
    const button = screen.getByRole("button");
    userEvent.click(button);
    expect(button).toBeInTheDocument();
    expect(mockedProps.clickToGame).toHaveBeenCalled();
  });

  it("button text contains prodided 'buttonNumber' prop content", () => {
    render(<LevelButton {...mockedProps} />);
    const button = screen.getByRole("button");
    console.log(button.innerText);
    expect(button).toHaveTextContent(mockedProps.buttonNumber);
  });
});
