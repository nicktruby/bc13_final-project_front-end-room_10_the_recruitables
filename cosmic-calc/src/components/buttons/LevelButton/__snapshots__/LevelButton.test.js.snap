// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LevelButton> successfully renders the component, matches Snapshot 1`] = `
<DocumentFragment>
  <div
    class="levelButton"
  >
    <button
      class="levelButtons"
      id="levelButton-1234"
    />
  </div>
</DocumentFragment>
`;

exports[`<LevelButton> successfully renders the component, matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="levelButton"
  >
    <button
      class="levelButtons"
      id="levelButton-1234"
    >
      BUTTON 1234
    </button>
  </div>
</DocumentFragment>
`;
